rootfs/etc/nginx/lua/util/dns.lua

```diff
@@ -28,48 +27,77 @@ local function a_records_and_max_ttl(answers)
   return addresses, ttl
 end
 
-function _M.resolve(host)
-  local cached_addresses = cache:get(host)
-  if cached_addresses then
-    local message = string.format(
-      "addresses %s for host %s was resolved from cache",
-      table.concat(cached_addresses, ", "), host)
-    ngx.log(ngx.INFO, message)
-    return cached_addresses
+local function format_ipv6_nameservers(nameservers)
+  local formatted_nameservers = {}
+  for _, nameserver in ipairs(nameservers) do
+    local formatted_nameserver = nameserver
+    if not nameserver:match("^%d+.%d+.%d+.%d+$") then
+      formatted_nameserver = string.format("[%s]", nameserver)
+    end
+    table.insert(formatted_nameservers, formatted_nameserver)
   end
+  return formatted_nameservers
+end
 
+local function resolve_host(host, qtype)
   local r
   r, err = resolver:new{
-    nameservers = util.deepcopy(configuration.nameservers),
+    nameservers = format_ipv6_nameservers(configuration.nameservers),
```

Comment:
you don't need this anymore as we do that in the controller side now

Reply:
done